src/common/associatedUtils.ts

```diff
@@ -0,0 +1,28 @@
+import { get, isEmpty } from 'lodash';
+
+const getResourceName = resource => get(resource, 'name.singular');
+const hasParent = parent => !isEmpty(parent);
+
+const isChildOfPolicy = parent => hasParent(parent) && isPolicy(parent);
+const isGroupPermission = (parent, child) =>
+  hasParent(parent) && isGroup(parent) && isPermission(child);
+const isUserPermission = (parent, child) =>
+  hasParent(parent) && isUser(parent) && isPermission(child);
+
+const isPolicy = resource => getResourceName(resource) === 'policy';
+const isPermission = resource => getResourceName(resource) === 'permission';
+const isUser = resource => getResourceName(resource) === 'user';
+const isGroup = resource => getResourceName(resource) === 'group';
```

Comment:
out of curiosity, are `Policy`, `Permission`, `User`, `Group` and `ApiKey` typescript types? if so, these functions can be typed as `(resource: Resource) => resource is Policy`. Probably will break things more than helpful for now so not needed, but just FYI :)

Reply:
There are ts types for the entities, but they are more of a DTO...the resource type here is from that giant resource_map file.